### PR TITLE
Add miscellaneous function displayName()

### LINKED ISSUE #36770: Add miscellaneous function `displayName()`
**Use case**

It is better than hostname in the following example:
```
SELECT hostname() AS host, event_time, round(value, 2) AS v, bar(v, 0, 10, 100) FROM system.asynchronous_metric_log WHERE event_date >= yesterday() AND event_time >= now() - 300 AND metric = 'LoadAverage1' ORDER BY host, event_time
```

Because hostnames can be non-informative like `ip-172-31-5-46`.

### PR DESCRIPTION
**Pull request by xlwh**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Add function `displayName`, closes #36770

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: tests/queries/0_stateless/02313_displayname_test.reference [hunk 1]
```diff
@@ -0,0 +1,2 @@
+localhost
+localhost
```
Comment (vdimir) [inline tests/queries/0_stateless/02313_displayname_test.reference:2]:
```suggestion
localhost

```

File: tests/queries/0_stateless/02313_displayname_test.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+select displayname();
+select displayName();
```
Comment (vdimir) [inline tests/queries/0_stateless/02313_displayname_test.sql:2]:
```suggestion
select displayName();

```

**Review by vdimir — COMMENTED** [review]
Thank you for your PR!

Maybe it's just worth to add some test case to integration tests, with distributed query and different `display_name`s in config. 

For example to this test:
https://github.com/ClickHouse/ClickHouse/blob/46fa87b14e5e8f3b7f4107157f5a33ef351ab0d2/tests/integration/test_settings_constraints_distributed/test.py#L101

**Comment by xlwh** [comment]
> 

ok, i'll add integration tests later. thanks. @vdimir

**Comment by alexey-milovidov** [comment]
@vdimir It's not necessary to use an integration test.
You can simply add a config for functional tests, see `tests/config` that will change the display name to `functional-tests`.

**Comment by alexey-milovidov** [comment]
@vdimir We don't need the `displayname` alias.

**Comment by alexey-milovidov** [comment]
Some small steps are still needed here...

**Pull request merged**